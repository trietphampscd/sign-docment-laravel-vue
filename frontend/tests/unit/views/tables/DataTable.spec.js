import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import DataTable from '@/views/tables/DataTable'


Vue.use(BootstrapVue)

describe('DataTable.vue', () => {
  it('has a name', () => {
    expect(DataTable.name).toMatch('DataTable')
  })
  it('has a created hook', () => {
    expect(typeof DataTable.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof DataTable.data).toMatch('function')
    const defaultData = DataTable.data()
    expect(defaultData.useVuex).toBe(false)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(DataTable)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is DataTable', () => {
    const wrapper = shallowMount(DataTable)
    expect(wrapper.is(DataTable)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(DataTable)
    expect(wrapper.find('div.card-header').text()).toMatch('Data Table')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(DataTable)
    expect(wrapper.element).toMatchSnapshot()
  })
})
