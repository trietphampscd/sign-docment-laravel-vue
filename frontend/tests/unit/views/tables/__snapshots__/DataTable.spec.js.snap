// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.vue renders correctly 1`] = `
<div
  classname="animated"
>
  <bcard-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    subtitletag="h6"
    tag="div"
    titletag="h4"
  >
    <bcardheader-stub
      headertag="div"
    >
      <i
        class="icon-menu mr-1"
      />
      Data Table
      
      <a
        class="badge badge-danger"
        href="https://coreui.io/pro/vue/"
      >
        CoreUI Pro
      </a>
       
      <div
        class="card-header-actions"
      >
        <a
          classname="card-header-action"
          href="https://github.com/matfish2/vue-tables-2"
          rel="noopener noreferrer"
          target="_blank"
        >
          <small
            classname="text-muted"
          >
            docs
          </small>
        </a>
      </div>
    </bcardheader-stub>
     
    <bcardbody-stub
      bodytag="div"
      subtitletag="h6"
      titletag="h4"
    >
      <v-client-table-stub
        columns="name,code,uri"
        data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        id="dataTable"
        options="[object Object]"
        theme="bootstrap4"
      />
    </bcardbody-stub>
  </bcard-stub>
</div>
`;
